fix: some scripts

elsongabriel opened the pull request:
# Description

Continues fixing of #1784 and #1847 

- Improved message on exercise_training;
- Fixes reward charges;
- Created flag in prison bosses' scripts to enable minus players and removed already created functions;
- Removed create torch when player login;
- Fixes in 'clearBossRoom' function usage on bosses teleports;
- Fixed cask_kegs (undo the last  pr);
- Fixed count days and time on ferumbras_lever and fixed mechanic of Ascending Ferumbras boss.
- Fixed xp boost storage resetter (on very ss, the storage was being presented, and the player always buy it with 30 tc);
- Fixed firewalker boots transform with 'worn leather boots' and a small enchanted ruby;
- Added 13.21 new mounts, reordered and fixed typos, changed the word moveable to movable ([grammarist link](https://grammarist.com/spelling/movable-moveable/#:~:text=Moveable%20is%20the%20older%20spelling,so%20for%20over%20a%20century.));
- Removed unnecessary spaces;

Please delete options that are not relevant.
  - [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested

**Test Configuration**:

  - Server Version: Canary 3.1.1
  - Client: Tibia 13.21
  - Operating System: Windows/Ubuntu

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings
  - [x] I have added tests that prove my fix is effective or that my feature works

elsongabriel committed:
exercise_training: improved msg.
broadcast: removed unnecessary Webhook.sendMessage.
reward: fixes charges.
prision bosses: created flag to enable minus players. removed functions already created.
login: removed create torch when player login.
bosses teleports: fixes in 'clearBossRoom' function usage.

elsongabriel committed:
cask_kegs: redo.

elsongabriel committed:
xp boost: removed the reset of xp boost storage on server save. (on every ss, the xp boost storage was cleaned)

elsongabriel committed:
firewalker boots transform: removed 'worn leather boots' to transforming when uses small enchanted ruby.

elsongabriel committed:
gamestore & mounts: added 13.21 new mounts, reordered and fixed typos, changed the word moveable to movable (https://grammarist.com/spelling/movable-moveable/#:~:text=Moveable%20is%20the%20older%20spelling,so%20for%20over%20a%20century.)

elsongabriel committed:
npc: added npc name in logs.

elsongabriel committed:
typo in movable: fixed typos, changed the word moveable to movable (https://grammarist.com/spelling/movable-moveable/#:~:text=Moveable%20is%20the%20older%20spelling,so%20for%20over%20a%20century.)

elsongabriel committed:
quests: removed unnecessary spaces.
broadcast: undo removed commented (Webhook.sendMessage) on broadcast, to resolve in further problems with encoding.

elsongabriel committed:
adjusted scripts.

elsongabriel committed:
reward: added message on reward item.

luan commented on data/scripts/talkactions/gm/broadcast.lua:
```diff
@@ -11,7 +11,7 @@ function broadcast.onSay(player, words, param)
 	local text = player:getName() .. " broadcasted: " .. param
 	logger.info(text)
-	Webhook.sendMessage("Broadcast", text, WEBHOOK_COLOR_WARNING, announcementChannels["serverAnnouncements"])
+	--Webhook.sendMessage("Broadcast", text, WEBHOOK_COLOR_WARNING, announcementChannels["serverAnnouncements"])
```
```suggestion
	Webhook.sendMessage("Broadcast", text, WEBHOOK_COLOR_WARNING, announcementChannels["serverAnnouncements"])
```

elsongabriel replied:
In a discussion with Edu, we decided to comment on this function because the webhook sendMessage function isn't encoding accentuation characters.

luan replied:
What happens? Tibia also has issues showing accents, but I'd still rather be able to broadcast webhooks in english

luan commented on data-canary/scripts/globalevents/global_server_save.lua:
```diff
@@ -10,8 +10,6 @@ local function ServerSave()
 	-- Updating daily reward next server save
 	UpdateDailyRewardGlobalStorage(DailyReward.storages.lastServerSave, os.time())
-	-- Reset gamestore exp boost count.
-	db.query("UPDATE `player_storage` SET `value` = 0 WHERE `player_storage`.`key` = 51052")
```
I think this is necessary to reset the exp boost price no?

elsongabriel replied:
yes. but only when the server starts up. with this reset query, on every SS (generally 1 per hour), the storage will be reset and the value will be every 30 tc (start price).

elsongabriel committed:
player.lua: removed unused function.
ferumbras_lever: fixed script and your summons creation.
soul war mount: created function to gives to the player the mount.
otservbr-custom-zones.xml: added to repository.

github-actions[bot] committed:
Lua code format - (Stylua)

elsongabriel committed:
Implemented luan review.

Co-authored-by: Luan Santos <github@luan.sh>

elsongabriel committed:
Merge branch 'main' into hotfix/fix-scripts

luanluciano93 commented on data/modules/scripts/gamestore/gamestore.lua:
```diff
@@ -2035,11 +2089,11 @@ GameStore.Categories = {
-				price = 870,
-				id = 164,
-				description = "{character}\n{speedboost}\n\n<i>A magical fire burns inside these wolves. Bred as the faithful guardians for an eccentric wizard's tower, these creatures make for loyal companions during your travels. While not originally intended for riding, their sturdy frame makes the Dawn Strayer, Dusk Pryer and Snow Strider suitable mounts.</i>",
+				icons = { "Spirit_of_Purity.png" },
```
essa mount não é da store.

luanluciano93 commented on data/modules/scripts/gamestore/gamestore.lua:
```diff
@@ -1498,6 +1526,14 @@ GameStore.Categories = {
 			},
+			{
+				icons = { "Foxmouse.png" },
+				name = "Foxmouse",
```
essa mount não é da store

luanluciano93 commented on data/modules/scripts/gamestore/gamestore.lua:
```diff
@@ -1306,6 +1316,14 @@ GameStore.Categories = {
 				type = GameStore.OfferTypes.OFFER_TYPE_MOUNT,
 			},
+			{
+				icons = { "Darkfire_Devourer.png" },
```
essa montaria é comprado no pacote "[Bundle](https://www.tibiawiki.com.br/wiki/Rotten_Blood_Bundle)."

elsongabriel replied:
tá funcionando essa compra? e vem essa mount ?

luanluciano93 replied:
teria que fazer essa aba e essa função de vir esse "pacote" de coisas, a mount junto.

elsongabriel committed:
Merge branch 'main' into hotfix/fix-scripts

elsongabriel committed:
set parameter boss position in checkBoss functions.

elsongabriel committed:
Merge branch 'main' into hotfix/fix-scripts

elsongabriel committed:
ferumbra ascendant quest: fixes.

dudantas commented on data-otservbr-global/scripts/actions/quests/ferumbras_ascendant/ferumbras_lever.lua:
```diff
@@ -1,30 +1,20 @@
-local config = {
+AscendingFerumbrasConfig = {
```
Isto está errado, essa tabela está como global, mas nada garante que ela vai ser carregada antes dos outros scripts serem lidos... Então, o que pode acontecer é que pode dar load antes nos outros scripts e depois aqui, o que vai retornar nil. Aconselho criar um arquivo "register_tables" na pasta scripts/libs e adicionar ela lá dentro, já fica para usos futuros o arquivo.

dudantas commented on data-otservbr-global/scripts/creaturescripts/quests/ferumbras_ascendant/ferumbras_soul_splinter.lua:
```diff
@@ -1,10 +1,15 @@
-	if not monster then
-		return true
-	end
+	Game.createMonster("ferumbras essence", creature:getPosition(), true, true)
```
Aconselho manter a verificação e adicionar um log, o "Game.createMonster" retorna um ponteiro e este ponteiro deve ser tratado.
```suggestion
	local monster = Game.createMonster("ferumbras essence", creature:getPosition(), true, true)
	if not monster then
		logger.error("[ferumbrasSoulSplinter.onDeath] cannot create monster on position {}", creature:getPosition():toString())
		return true
	end
```

dudantas commented on data-otservbr-global/scripts/globalevents/vip/online_coins.lua:
```diff
@@ -31,7 +31,7 @@ function onlineCoinsEvent.onThink(interval)
 	local checkIp = {}
 	for _, player in pairs(players) do
-		if player:getAccountType() >= ACCOUNT_TYPE_GAMEMASTER then
+		if player:getGroup():getId() >= GROUP_TYPE_SENIORTUTOR then
```
Creio que seja só ">" já que o senior tutor é apenas um jogador comum.
```suggestion
		if player:getGroup():getId() > GROUP_TYPE_SENIORTUTOR then
```

dudantas replied:
```suggestion
		if player:getGroup():getId() > GROUP_TYPE_SENIORTUTOR then
```

dudantas commented on data-otservbr-global/scripts/globalevents/vip/online_tokens.lua:
```diff
@@ -32,7 +32,7 @@ function onlineTokensEvent.onThink(interval)
 	local checkIp = {}
 	for _, player in pairs(players) do
-		if player:getAccountType() >= ACCOUNT_TYPE_GAMEMASTER then
+		if player:getGroup():getId() >= GROUP_TYPE_SENIORTUTOR then
```
```suggestion
		if player:getGroup():getId() > GROUP_TYPE_SENIORTUTOR then
```

dudantas commented on data/libs/functions/player.lua:
```diff
@@ -554,28 +554,6 @@ function Player.updateHazard(self)
 	return true
 end
 
-function Player:addItemStoreInbox(itemId, amount, moveable)
```
Apesar de não ser usada é uma função muito útil e que pode estar sendo usada por alguém que usa o projeto, remover ela vai também remover a compatibilidade... O que deve ser evitado, a não ser que haja um bom motivo.

dudantas commented on data-otservbr-global/scripts/actions/quests/ferumbras_ascendant/ferumbras_lever.lua:
```diff
@@ -83,16 +73,15 @@ function ferumbrasAscendantLever.onUse(player, item, fromPosition, target, toPos
 						local position = Position(33392 + xrand, 31473 + yrand, 14)
-						if Game.createMonster("rift invader", position) then
-						end
+						Game.createMonster(config.summonName, position)
```
```suggestion
						if not Game.createMonster(config.summonName, position) then
							logger.error("[ferumbrasAscendantLever.onUse] can't create monster {}, on position {}", config.summonName, position:toString())
						end
```

elsongabriel committed:
Merge branch 'main' into hotfix/fix-scripts

elsongabriel committed:
Edu suggestion.

Co-authored-by: Eduardo Dantas <eduardo.dantas@hotmail.com.br>

elsongabriel committed:
Edu suggestion.

Co-authored-by: Eduardo Dantas <eduardo.dantas@hotmail.com.br>

elsongabriel committed:
Edu suggestion.

Co-authored-by: Eduardo Dantas <eduardo.dantas@hotmail.com.br>

elsongabriel committed:
Re-added function.
Moved global config to a register lib.

elsongabriel committed:
Fix vortex.

github-actions[bot] committed:
Lua code format - (Stylua)

elsongabriel committed:
Merge branch 'main' into hotfix/fix-scripts

dudantas committed:
fix: wrong condition

dudantas committed:
fix: final changes

dudantas committed:
Merge branch 'main' into pr/1821

macvirii commented:
I really like the /tp solution of this PR. But I still think there should be a way to remove the TP too... Or am I being unreasonable.
